pkg/fleet-manager/fleet_plugin.go

```diff
@@ -0,0 +1,492 @@
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secret.Namespace,
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return nil
+		}
+	} else if err != nil {
+		return nil
+	}
+
+	s, err := kclient.KubeClient().CoreV1().Secrets(secret.Namespace).Get(ctx, secret.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err := kclient.KubeClient().CoreV1().Secrets(secret.Namespace).Create(ctx, secret, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
+
+	s.Data = secret.Data
+	_, err = kclient.KubeClient().CoreV1().Secrets(secret.Namespace).Update(ctx, s, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (f *FleetManager) reconcileGrafanaPlugin(ctx context.Context, fleet *fleetapi.Fleet) (kube.ResourceList, error) {
+	// TODO: Implement me
+	return nil, nil
+}
+
+func (f *FleetManager) reconcilePluginResources(ctx context.Context, fleet *fleetapi.Fleet, resources kube.ResourceList) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log = log.WithValues("fleet", types.NamespacedName{Name: fleet.Name, Namespace: fleet.Namespace})
+
+	repoDict, releaseDict := sets.New[types.NamespacedName](), sets.New[types.NamespacedName]()
+	for _, res := range resources {
+		switch res.Mapping.GroupVersionKind.Kind {
+		case sourcev1.HelmRepositoryKind:
+			repoDict.Insert(types.NamespacedName{Namespace: res.Namespace, Name: res.Name})
+		case hrapiv2b1.HelmReleaseKind:
+			releaseDict.Insert(types.NamespacedName{Namespace: res.Namespace, Name: res.Name})
+		default:
+			// should not happen, but just in case
+			log.V(2).Info("unexpected resource type", "kind", res.Mapping.GroupVersionKind.Kind)
+		}
+	}
+
+	fleetOwnerRef := metav1.OwnerReference{
+		APIVersion: fleetapi.GroupVersion.String(),
+		Kind:       "Fleet",
+		Name:       fleet.Name,
+		UID:        fleet.UID,
+	}
+
+	helmRepos := &sourcev1.HelmRepositoryList{}
+	helmReleases := &hrapiv2b1.HelmReleaseList{}
+	fleetLabels := fleetResourceLables(fleet.Name)
+	if err := f.Client.List(ctx, helmRepos, client.InNamespace(fleet.Namespace), fleetLabels); err != nil {
+		log.Error(err, "failed to list helm repositories")
+		return ctrl.Result{}, err
+	}
+	if err := f.Client.List(ctx, helmReleases, client.InNamespace(fleet.Namespace), fleetLabels); err != nil {
+		log.Error(err, "failed to list helm repositories")
+		return ctrl.Result{}, err
+	}
+
+	for _, repo := range helmRepos.Items {
+		if !repoDict.Contains(types.NamespacedName{Namespace: repo.Namespace, Name: repo.Name}) {
+			// delete redundant helm releases
+			if err := f.Client.Delete(ctx, &repo); err != nil {
+				log.Error(err, "failed to delete helm repository")
+				return ctrl.Result{}, err
+			}
+		}
+
+		// ensure owner reference
+		repo.OwnerReferences = capiutil.EnsureOwnerRef(repo.OwnerReferences, fleetOwnerRef)
+		if err := f.Client.Update(ctx, &repo); err != nil {
```

Comment:
where do you create the resources

Reply:
HelmRelease and HelmRepo are created in `reconcileMetricPlugin`